packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,340 @@
+import { BigDecimal } from './big_decimal';
+import { sha3 } from './hash';
+import { Address, Bytes, HexString, Numbers, ValueTypes } from './types';
+import { isAddress, isHexStrict } from './validation';
+
+const base = BigInt(10);
+const expo10 = (expo: number) => base ** BigInt(expo);
+
+// Ref: https://ethdocs.org/en/latest/ether.html
+/** @internal */
+const ethUnitMap = {
+	noether: BigInt('0'),
+	wei: BigInt(1),
+	kwei: expo10(3),
+	Kwei: expo10(3),
+	babbage: expo10(3),
+	femtoether: expo10(3),
+	mwei: expo10(6),
+	Mwei: expo10(6),
+	lovelace: expo10(6),
+	picoether: expo10(6),
+	gwei: expo10(9),
+	Gwei: expo10(9),
+	shannon: expo10(9),
+	nanoether: expo10(9),
+	nano: expo10(9),
+	szabo: expo10(12),
+	microether: expo10(12),
+	micro: expo10(12),
+	finney: expo10(15),
+	milliether: expo10(15),
+	milli: expo10(15),
+	ether: expo10(18),
+	kether: expo10(21),
+	grand: expo10(21),
+	mether: expo10(24),
+	gether: expo10(27),
+	tether: expo10(30),
+};
+
+const errorMessage = (value: unknown, error: string) =>
+	`Invalid value given "${String(value)}". Error: ${error}.`;
+
+export type EtherUnits = keyof typeof ethUnitMap;
```

Comment:
What's the benefit of doing it this way as opposed to an enum?

Reply:
With the enum values, user has to pass the enum. With this approach user can pass matching `string` values. Both approaches are type safe, but this one seems more friendly for user.